### PR TITLE
feat(page): campaigns list

### LINKED ISSUE #230: Campaign List Page
## User Story
As user I want to browse and explore campaigns so I can find interesting projects on GameDAO.

## Acceptance Criteria

1. Clicking on the "Campaigns" link in the top navi leads the user to the campaigns list page
2. The Title "Campaigns" is on top of the page
3. Some top list elements are available like 
3.1 search bar is which can search in campaign titles and organisation names
3.2 a create campaign button (do not link yet)
3.3 a filter option element
3.4 a sorting element
4. A funding target filter is available with following options (all checkboxes) (refer to "Funding targets table" in additional information)
5. Campaign status filter is available with following (options all checkboxes) (refer to "Campaign status table" in additional information
6. Categories filter with all available categories (all checkboxes)
7. Sorting is possible by (all ascending, descending)
7.1 funding target 
7.2 current funding 
7.3 contributors count  
7.4 alphabetical,
7.5 time left 
8. Campaign Cards are displayed as provided in Figma designs (see additional information)
9. A click on the Campaign Card Image and Title links to the Campaign Detail Page

### **Funding targets table** 
| Funding targets      |
|---------------------|
| < 1,000 $            |
| 1,000 $ - 9,999 $     |
| 10,000 $ - 99,999 $   |
| 100,000 $ - 499,999 $ |
| > 500,000 $          |

### **Campaign status table** 
|   Status   | StatusID |  UI Status  |  Color | Meaning                                                                                                                                   |
|:----------:|:--------:|:-----------:|:------:|-------------------------------------------------------------------------------------------------------------------------------------------|
| Init       | 0        | Not Started | blue   | campaign created but not started yet                                                                                                      |
| Active     | 1        | Active      | yellow | campaign and active. Open for funding.                                                                                                    |
| Paused     | 2        | Paused      | blue   | campaign paused, eg. through GameDAO council vote                                                                                         |
| Finalizing | 3        | Finalizing  | green  | campaign ended, funded and contributions are being processed (eg. from reserved to transfer to DAO treasury)                              |
| Reverting  | 4        | Reverting   | red    | campaign ended, funding goal not reached and contributions are being processed (eg. from reserved to unreserved)                          |
| Success    | 5        | Funded      | green  | campaign ended, funded and contributions are already processed (funds are locked in DAO treasury)                                         |
| Failed     | 6        | Failed      | red    | campaign ended, funding goal not reached and contributions are already processed (all contributions are unreserved / transferrable again) |
| Locked     | 7        | Locked      | red    | campaign locked (eg, through GameDAO council vote)                                                                                        |

Figma: https://www.figma.com/file/8xFCnDl6SlUOonFm1V20ne/GameDAO-Haiku-UI?node-id=616%3A41740

## How to test

1. a
2. b
3. c
9. d

### PR DESCRIPTION
**Pull request by developer-sidani**
### Done:
- Fetch Campaigns
- Display Campaigns
- Use translation for buttons and texts
- Apply pagination (only viewable when campaigns are over 15)
- Add search Bar
- Merge with main after PR for the campaigns list refactor
- Generalize the filters section for both organizations and campaigns page
- Filter query according to user input
- Fetch the campaign sort options and the campaign filters from GraphQL

### PS:
- Filters for funding targets are removed from this task upon approval
- Sort options, filters, and search are managed by GraphQL

### CODE REVIEW

**Comment by developer-sidani** [comment]
> @developer-sidani please think of fallback images and can you avoid having campaign cards with different sizes?: image

@5-mark This should be managed through the ticket responsible for the layout of the campaign card #247

**Comment by developer-sidani** [comment]
PS: after discussion with @5-mark @2075 @DarkNebula0, there is not going to be filters for the target funded for this ticket, so definitely it is going to be managed by another ticket

File: pkg/app/src/pages/campaigns/index.tsx [hunk 1]
```diff
@@ -1,97 +1,164 @@
...
+	})
+	const paginatedData = useMemo<Campaign[]>(() => data?.campaign?.slice() as Campaign[], [data])
+	const { t } = useTranslation()
+	const buttonVisibility = useMemo(
+		() => paginatedData?.length < campaignsCount?.data?.campaign_aggregate?.aggregate?.count,
+		[paginatedData?.length, campaignsCount?.data],
+	)
 	return (
-		<Layout showHeader showFooter showSidebar title="Campaigns">
+		<Layout showHeader showFooter showSidebar title="Organisation">
```
Comment (DarkNebula0) [inline pkg/app/src/pages/campaigns/index.tsx:91]: Organization on the campaign site?

File: pkg/app/src/@types/campaign.d.ts [hunk 2]
```diff
@@ -8,3 +10,13 @@ export interface TmpCampaignState extends TmpCampaign {
 	setDescription: (string) => void
 	clearAll: () => void
 }
+export interface CampaignsSortOptions {
```
Comment (DarkNebula0) [inline pkg/app/src/@types/campaign.d.ts:13]: Weren't they already created by graphql?

**Review by DarkNebula0 — CHANGES_REQUESTED** [review]

**Comment by 5-mark** [comment]
@developer-sidani is it possible to set the default state of the filters like this:
image
and the sorting default state for time left (the ones where least time is left on the top of the page, but not the ones where time is up already)

**Review by DarkNebula0 — APPROVED** [review]

**Pull request merged**